### PR TITLE
Prework - Task 5d - Item class

### PR DESCRIPTION
**Pull request by mariuszlisowski**

### CODE REVIEW

File: shm/src/Item.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    return *this;
+}
+
+Cargo& Item::operator-=(size_t amount) {
+    if (amount <= amount_) {
+        amount_ -= amount;
+    }
+
+    return *this;
```
Comment (lukasz-kukulka) [inline shm/src/Item.cpp:28]: Redundant line

File: shm/src/Item.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+std::string Item::getName() const{
+    return name_;
```
Comment (lukasz-kukulka) [inline shm/src/Item.cpp]: Space after const

File: shm/src/Item.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+Item::Item(const std::string& name, size_t amount, size_t basePrice, Rarity rarity)
+    : Cargo(name, amount, basePrice), rarity_(rarity)
+{}
+
+bool Item::operator==(const Cargo& cargo) const {
+    if (typeid(cargo) == typeid(Item)) {
+        const Item* item = static_cast<const Item*>(&cargo);
+        return name_ == item->getName() && amount_ == item->getAmount() && basePrice_ == item->getBasePrice() &&
+               rarity_ == item->getRarity();
```
Comment (lukasz-kukulka) [inline shm/src/Item.cpp]: Im not sure if we need compare amout, amout don't need be the same.

File: shm/src/Item.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    return false;
```
Comment (lukasz-kukulka) [inline shm/src/Item.cpp:14]: Redundant line

**Review by lukasz-kukulka — APPROVED** [review]

File: shm/src/Item.cpp (hunk 3 above)
Comment (mariuszlisowski) [reply shm/src/Item.cpp]: It depends but I'll remove it for now.

File: shm/src/Item.cpp (hunk 2 above)
Comment (mariuszlisowski) [reply shm/src/Item.cpp]: Ok, done.

File: shm/src/Item.cpp (hunk 1 above)
Comment (mariuszlisowski) [reply shm/src/Item.cpp:28]: newline-before-return rule:
This rule requires an empty line before return statements to increase code clarity, except when the return is alone inside a statement group (such as an if statement). In the latter case, the return statement does not need to be delineated by virtue of it being alone.

File: shm/src/Item.cpp [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        const Item* item = static_cast<const Item*>(&cargo);
+        return name_ == item->getName() && basePrice_ == item->getBasePrice() &&
+               rarity_ == item->getRarity();
+    }
+
+    return false;
+}
+
+Cargo& Item::operator+=(size_t amount) {
+    amount_ += amount;
```
Comment (BarTes8) [inline shm/src/Item.cpp:18]: should check capacity

File: shm/src/Item.cpp [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+    return *this;
+}
+
+Cargo& Item::operator-=(size_t amount) {
+    if (amount <= amount_) {
+        amount_ -= amount;
```
Comment (BarTes8) [inline shm/src/Item.cpp:25]: maybe we should check if amount > amount_ and return *this; ?

File: shm/src/Item.cpp [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    return *this;
+}
+
+std::string Item::getName() const {
```
Comment (BarTes8) [inline shm/src/Item.cpp:31]: add override

File: shm/src/Item.cpp [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    return name_;
+}
+
+size_t Item::getAmount() const {
```
Comment (BarTes8) [inline shm/src/Item.cpp:35]: add override

File: shm/src/Item.cpp [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    return amount_;
+}
+
+size_t Item::getBasePrice() const {
```
Comment (BarTes8) [inline shm/src/Item.cpp:39]: add override

**Review by lukasz-kukulka — APPROVED** [review]

**Pull request closed without merging**